### PR TITLE
feat(function): Support Semi-structured function json_extract_path_text

### LINKED ISSUE #4559: Feature: Semi-structured data function json_extract_path_text
**Summary**

Add support Semi-structured Data Function `JSON_EXTRACT_PATH_TEXT` to get the value pointed by the path.

https://docs.snowflake.com/en/sql-reference/functions/json_extract_path_text.html

### PR DESCRIPTION
**Pull request by b41sh**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Support `JSON_EXTRACT_PATH_TEXT` function

## Changelog

- New Feature
- Build/Testing/CI
- Documentation

## Related Issues

Fixes #4559

### CODE REVIEW

File: common/functions/src/scalars/semi_structureds/json_extract_path_text.rs [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        } else {
+            return NullType::arc().create_constant_column(&DataValue::Null, input_rows);
+        }
+        let variant_column = builder.build(input_rows);
+        let column = extract_value_by_path(&variant_column, path_keys, input_rows, false)?;
+
+        // convert VariantColumn to StringColumn
+        let (is_all_null, valids) = column.validity();
+        if is_all_null {
+            return NullType::arc().create_constant_column(&DataValue::Null, input_rows);
```
Comment (sundy-li) [inline common/functions/src/scalars/semi_structureds/json_extract_path_text.rs:96]: This is inconsistent with `return_type`.

Comment (b41sh) [reply common/functions/src/scalars/semi_structureds/json_extract_path_text.rs]: I have removed this.

File: common/functions/src/scalars/semi_structureds/json_extract_path_text.rs [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+            match serializer.serialize_json_object(columns[0].column(), None) {
+                Ok(values) => {
+                    for v in values {
+                        builder.append(&VariantValue::from(v));
+                    }
+                }
+                Err(e) => return Err(e),
+            }
+        } else {
+            return NullType::arc().create_constant_column(&DataValue::Null, input_rows);
```
Comment (sundy-li) [inline common/functions/src/scalars/semi_structureds/json_extract_path_text.rs:88]: ditto

Comment (b41sh) [reply common/functions/src/scalars/semi_structureds/json_extract_path_text.rs]: done.

**Review by sundy-li — APPROVED** [review]

**Pull request merged**